currency-converter/currency.js

```diff
@@ -1,80 +1,163 @@
+
+      console.log("Inserted new rate:", currencyRates);
+      const insertConfirmation = document.getElementById("insertConfirmation");
+      insertConfirmation.textContent = `Inserted new rate: 1 ${baseCurrency} = ${rate} ${targetCurrency}`;
+      insertConfirmation.style.color = "black";
+      displayRates(currencyRates);
+    });
+
+  document.getElementById("convertForm").addEventListener("submit", (event) => {
     event.preventDefault();
     const amount = parseFloat(document.getElementById("amount").value);
     const fromCurrency = document
       .getElementById("fromCurrency")
-      .value.toUpperCase.trim();
+      .value.toUpperCase()
+      .trim();
     const toCurrency = document
       .getElementById("toCurrency")
-      .value.toUpperCase.trim();
+      .value.toUpperCase()
+      .trim();
 
-    if (rates.base === fromCurrency && rates.rates[toCurrency]) {
-      const convertedAmount = amount * rates.rates[toCurrency];
+    const rateObj = currencyRates.find((rate) => rate.base === fromCurrency);
+    if (rateObj && rateObj.rates[toCurrency]) {
+      const convertedAmount = amount * rateObj.rates[toCurrency];
+      const conversionResult = document.getElementById("conversionResult");
       conversionResult.textContent = `${amount} ${fromCurrency} = ${convertedAmount.toFixed(
         2
       )} ${toCurrency}`;
     } else {
+      const conversionResult = document.getElementById("conversionResult");
       conversionResult.textContent = "Conversion rate not available.";
     }
   });
 
-  updateRateForm.addEventListener("submit", (event) => {
-    event.preventDefault();
-    const baseCurrency = document
-      .getElementById("updateBaseCurrency")
-      .value.toUpperCase.trim();
-    const targetCurrency = document
-      .getElementById("updateTargetCurrency")
-      .value.toUpperCase.trim();
-    const rate = parseFloat(document.getElementById("updateRate").value);
-
-    if (rates.base === baseCurrency) {
-      rates.rates[targetCurrency] = rate;
-      console.log("Updated rate:", rates);
-      updateConfirmation.textContent = `Updated rate: 1 ${baseCurrency} = ${rate} ${targetCurrency}`;
-      updateConfirmation.style.color = "green";
-    } else {
-      alert(
-        updateConfirmation.textContent = `Base currency does not match the existing base currency: ${rates.base}`;
-      updateConfirmation.style.color = "black";
-      );
-    }
-  });
+  document
+    .getElementById("updateRateForm")
+    .addEventListener("submit", (event) => {
+      event.preventDefault();
+      const baseCurrency = document
+        .getElementById("updateBaseCurrency")
+        .value.toUpperCase()
+        .trim();
+      const targetCurrency = document
+        .getElementById("updateTargetCurrency")
+        .value.toUpperCase()
+        .trim();
+      const rate = parseFloat(document.getElementById("updateRate").value);
+
+      const rateObj = currencyRates.find((rate) => rate.base === baseCurrency);
```

Comment:
This is a good start for finding a rate when the `base` and `to` currencies are provided. However, the code needs to also work when only one currency is provided. Have a look at my messages on Slack. You can either make a search function with two arguments like [this](https://hackyourfuture-cph.slack.com/archives/C07AT59R9QR/p1720790626662359) or have only one argument like [this](https://hackyourfuture-cph.slack.com/archives/C07AT59R9QR/p1720794065032809).

Reply:
i have changed the search input to display the available rates even if one of the fields id empty.